Example/Sources/ConversationViewController.swift

```diff
@@ -0,0 +1,106 @@
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+        
+        let msg5 = "Lorem ipsum dolor sit amet, consectetur adipiscing elit." +
+                   "Pellentesque venenatis, ante et hendrerit rutrum" +
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+    }
+    
+    func currentSender() -> Sender {
+        return Sender(id: "123", displayName: "Steven")
+    }
+    
+    func numberOfMessages(in collectionView: UICollectionView) -> Int {
+        return messages.count
+    }
+    
+    func messageForItem(at indexPath: IndexPath, in collectionView: UICollectionView) -> MessageType {
+        return messages[indexPath.section]
+    }
+    
+    func avatarForMessage(_ message: MessageType, at indexPath: IndexPath, in collectionView: UICollectionView) -> Avatar {
+        let image = isFromCurrentSender(message: message) ? #imageLiteral(resourceName: "Steve-Jobs") : #imageLiteral(resourceName: "Tim-Cook")
+        return Avatar(placeholderImage: image)
```

Comment:
@lzyraining Thanks, I had considered making `Avatar` optional here as well. It will take a bit of time to decide on a nice API. The great thing is we don't have to worry about API stability at this moment 😃 

Thanks for raising the concerns of having to build the models required by protocols. I was originally thinking that people would probably already have a `User` model that they could conform `Sender` to -- but I see this could also be difficult and intrusive.